Source/WebCore/platform/graphics/ca/TileController.h

```diff
@@ -288,6 +291,7 @@ class TileController final : public TiledBacking {
     float m_tileDebugBorderWidth { 0 };
     ScrollingModeIndication m_indicatorMode { SynchronousScrollingBecauseOfLackOfScrollingCoordinatorIndication };
     FloatBoxExtent m_obscuredContentInsets;
+    FloatBoxExtent m_activeObscuredInsetsDelta;
```

Comment:
What's the life time of this? Should it be a std::optional<> and be unset when the state stabilizes?

Reply:
It's "unset" in the sense that it becomes a zero sized extent when the state stabilizes.

Reply:
i.e. instead of `view->obscuredInsetsDidChange(insets - m_obscuredInsets);` we'd call `view->obscuredInsetsDidChange(std::nullopt);` when the inset no longer changes.

Reply:
(and the lifetime is one visible rect update cycle after unobscured inset assignment)

Reply:
Made the life time more obvious by using a `std::optional<>` and by providing an explicit unset path `TiledBacking::clearObscuredInsetsAdjustments()`